### ISSUE TITLE
Bug introduced in `bmcsetup` by previous commit

### ISSUE THREAD

**Issue by samveen** [xcat2/xcat-core#3583]
Commit bb460dc5ceaa906c485a523de307840e4c639363 introduced a minor bug, where the value of `LOCKEDUSERS` was no longer considered while handling existing entries in the user list.

Line 432 is
```
if [ "$USERSLOT" == 0 ]; then USERSLOT=2; fi
```
while it should be
```
if [ "$USERSLOT" == 0 -o -z "$LOCKEDUSERS" ]; then USERSLOT=2; fi
```

**Comment by zet809** [comment]
Hi, @samveen , do you know why need to lock user for  ""$IPMIMFG" == 19046 -a "$XPROD" == 13616"?

**Comment by samveen** [comment]
@zet809 The code change I introduced only triggers in case values of $IPMICFG and $XPROD are not recognized from among the listed values.

For us, we want the `bmcsetup` script to overwrite the factory pre-configured user instead of disabling it. In xCAT `2.9.1`, the overwrite code-path triggered when `LOCKEDUSERS` was not defined (unrecognized $IPMICFG and $XPROD). However, now that code-path is missing, and from the description of  commit bb460dc5ceaa906c485a523de307840e4c639363 ` Provide a more adaptive approach to 'LOCKEDUSERS' in bmcsetup`, I can see that instead of an adaptive approach, the value LOCKEDUSERS became unused: 
```
[samveen@development xcat-core]$ git describe
2.13.5
[samveen@development xcat-core]$ grep LOCKEDUSERS xCAT-genesis-scripts/bin/bmcsetup 
        LOCKEDUSERS=1
        LOCKEDUSERS=1
   LOCKEDUSERS=1
        LOCKEDUSERS=1
    LOCKEDUSERS=1
```

Compare this to xCAT 2.11.1, for example:
```
[samveen@development xcat-core]$ git describe
2.11.1
[samveen@development xcat-core]$ grep LOCKEDUSERS xCAT-genesis-scripts/bin/bmcsetup 
        LOCKEDUSERS=1
        LOCKEDUSERS=1
        LOCKEDUSERS=1
    LOCKEDUSERS=1
if [ ! -z "$LOCKEDUSERS" ]; then
```

**Comment by zet809** [comment]
OK, thx!

**Comment by zet809** [comment]
I will close this issue since the code had been merged and will be included in 2.13.6 release.
